website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
The basic point that Jeff wants confirmation on is that the RoleSessionName on roles assumed by EC2 instances is indeed the instance ID. AWS implies as much in their [documentation](http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html#principaltable) -- "Assumed role" shows caller-specified-role-name, while "Role assigned to an Amazon EC2 instance" shows that as being the ec2-instance-id.

I have a hard time seeing Amazon change this, but I'll reach out to our Amazon support guys to see if I can get a good answer on their viewpoint on changing this (though I'm not sure I'd be able to share the details publicly).